src/containers/News/NewsPage.js

```diff
@@ -0,0 +1,139 @@
+import React, { Component } from 'react';
+import {
+  Link, withRouter, Switch, Route,
+} from 'react-router-dom';
+import MDSpinner from 'react-md-spinner';
+import { connect } from 'react-redux';
+import Fire from '../../config/fire';
+import classes from './News.module.scss';
+import AddEdit from './AddEdit/AddEdit';
+import NewsItem from './DisplayNews';
+import Button from '../../commonComponents/Button/Button';
+import BtnClass from '../../commonComponents/Button/Button.module.scss';
+
+class NewsPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      body: '',
+      edit: false,
+      spinner: false,
+    };
+  }
+
+  componentDidMount() {
+    const { user: { uid } } = this.props;
+    this.setState({ spinner: true });
+    const database = Fire.database();
+    const ref = database.ref('news');
+    ref.orderByChild('submittedBy').equalTo(uid).on('value', this.gotData, this.errData);
+  }
+
+  editHandler = (titleVal, bodyVal) => {
+    this.setState({
+      title: titleVal,
+      body: bodyVal,
+      edit: true,
+    });
+  }
+
+  gotData = (data) => {
+    if (data.val() == null) {
+      this.setState({ spinner: true });
```

Comment:
spinner should be false here, as you have already received the result from server here, which is null, but the operation is complete. you can maybe, show a note like 'No items yet' to the user here.